package.json

```diff
@@ -21,24 +25,25 @@
   "homepage": "https://carbon-addons-iot-react.com",
   "scripts": {
     "build": "cross-env BABEL_ENV='production' NODE_ENV='production'  yarn build:pre && rollup -c && yarn build:post",
-    "build:pre": "rimraf lib es umd css scss tmp storybook-static",
+    "build:pre": "yarn test:engines && rimraf lib es umd css scss tmp storybook-static",
     "build:post": "rimraf tmp",
-    "build:storybook": "NODE_OPTIONS='--max-old-space-size=16384' build-storybook -s public/production",
+    "build:storybook": "yarn test:engines && NODE_OPTIONS='--max-old-space-size=16384' build-storybook -s public/production",
     "format": "prettier --write \"**/*.{scss,css,js,jsx,md,ts}\"",
     "format:diff": "prettier --list-different \"**/*.{scss,css,js,jsx,md,ts}\"",
     "lint": "yarn lint:javascript && yarn lint:stylelint",
     "lint:javascript": "eslint --ext .jsx --ext .js .",
     "lint:stylelint": "stylelint './src/**/*.scss' --syntax scss --ignorePath .gitignore",
     "prepare": "yarn build",
     "publish-npm": "yarn semantic-release",
-    "start": "yarn storybook",
-    "storybook": "start-storybook -p 3000 -s public/development",
-    "test": "yarn test:a11y && yarn test:base",
-    "test:base": "cross-env TZ=America/Chicago jest --testPathPattern='.*\\.test\\.js(x)?' --coverage",
-    "test:a11y": "cross-env TZ=America/Chicago jest --forceExit --testPathPattern='.*\\.a11y\\.js(x)?'",
-    "test:watch": "cross-env TZ=America/Chicago jest --testPathPattern='.*\\.test\\.js(x)?' --watch --verbose --coverage",
-    "test:update": "cross-env TZ=America/Chicago jest src/components/StorybookSnapshots.test.js src/utils/__tests__/publicAPI.test.js --updateSnapshot",
-    "test:clear": "cross-env TZ=America/Chicago jest --clearCache"
+    "start": "yarn test:engines && yarn storybook",
+    "storybook": "yarn test:engines && start-storybook -p 3000 -s public/development",
+    "test": "yarn test:engines && yarn test:a11y && yarn test:base",
+    "test:base": "yarn test:engines && cross-env TZ=America/Chicago jest --testPathPattern='.*\\.test\\.js(x)?' --coverage",
+    "test:a11y": "yarn test:engines && cross-env TZ=America/Chicago jest --forceExit --testPathPattern='.*\\.a11y\\.js(x)?'",
```

Comment:
is this going to run `test:engines` three times when we run `yarn:test`?

Reply:
@davidicus Yes. It's a cheap/fast assertion. I want to ensure that engines are valid even if only a subset of the tests are ran. Trying to cover all bases - we've ran into too many issues from this.